### PR TITLE
Allow passing height and width as string in `Blocks.svelte`

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Allows height and width to be passed as a string into `Blocks.svelte`, thus allowing height to be set as a percentage (or any unit available for height). This will apply to any components that accept `height` 👀 

Closes: #4966

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/atoms/src/Block.svelte [hunk 1]
```diff
@@ -27,7 +38,7 @@
 	class:padded={padding}
 	class:border_focus={border_mode === "focus"}
 	class:hide-container={!explicit_call && !container}
-	style:height={typeof height === "number" ? height + "px" : undefined}
+	style:height={getHeight()}
```
Comment (aliabid94) [inline js/atoms/src/Block.svelte:41]: I don't believe this is reactive, e.g. if height changes this won't update. `height` should be an argument to this function to make it reactive

**Comment by aliabid94** [comment]
Didn't work for me:

```python
with gr.Blocks() as demo:
    name = gr.Chatbot(value=[["1", "2"]], height="100%")
    greet_btn = gr.Button("Test")
```

shows
Screen Shot 2023-11-24 at 11 01 02 AM

Also I think it's somewhat unclear what the percentage would be relative to.

**Comment by abidlabs** [comment]
> This should actually be documented for clarity. Maybe I'll update the Python docstring maybe? Open to thoughts

Agreed about documentation. It might be a lot to squeeze into the docstrings. What about adding a section in this Guide: https://www.gradio.app/guides/controlling-layout

File: gradio/components/chatbot.py
Comment (dawoodkhan82) [inline gradio/components/chatbot.py:1]: We should update ` height: int | None = None,` to include `string` type. Also should we update the other components that have a `height` param?

**Review by dawoodkhan82 — CHANGES_REQUESTED** [review]

**Review by pngwn — APPROVED** [review]
I would echo @dawoodkhan82's comment about the docstrings but other than that looks great. Thanks @hannahblair!

**Review by aliabid94 — APPROVED** [review]
LGTM!

**Review by pngwn — APPROVED** [review]
Looks good, thanks for taking care of the improvements @hannahblair.

Feel free to dismiss the stale review and merge.

**Review by pngwn — COMMENTED** [review]
tweaks for formatting

**Pull request merged**